### ISSUE TITLE
SO4 issue

### ISSUE THREAD

**Issue by yanxon** [qzhu2017/PyXtal_FF#30]
@David-Zagaceta 

I found this error while using the SO4 descriptor. I believe `try/except` can't be used in `nopython` mode.

**Comment by David-Zagaceta** [comment]
@yanxon can you provide up the error message and code to reproduce the error?

**Comment by yanxon** [comment]
@David-Zagaceta 

You can use this script: https://github.com/qzhu2017/PyXtal_FF/blob/master/examples/UCSD-Si-214-25/so4-NN.py

The error is the following:
```
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/yanxon/anaconda3/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/yanxon/Desktop/PyXtal_FF/pyxtal_ff/utilities/__init__.py", line 173, in compute
    from pyxtal_ff.descriptors.SO4 import SO4_Bispectrum
  File "/home/yanxon/Desktop/PyXtal_FF/pyxtal_ff/descriptors/SO4.py", line 365, in <module>
    @nb.njit(nb.f8(nb.f8, nb.f8), cache=True, fastmath=True, nogil=True)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/decorators.py", line 186, in wrapper
    disp.compile(sig)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/dispatcher.py", line 693, in compile
    cres = self._compiler.compile(args, return_type)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/dispatcher.py", line 76, in compile
    status, retval = self._compile_cached(args, return_type)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/dispatcher.py", line 90, in _compile_cached
    retval = self._compile_core(args, return_type)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/dispatcher.py", line 108, in _compile_core
    pipeline_class=self.pipeline_class)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 972, in compile_extra
    return pipeline.compile_extra(func)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 390, in compile_extra
    return self._compile_bytecode()
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 903, in _compile_bytecode
    return self._compile_core()
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 890, in _compile_core
    res = pm.run(self.status)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 266, in run
    raise patched_exception
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 257, in run
    stage()
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 404, in stage_analyze_bytecode
    func_ir = translate_stage(self.func_id, self.bc)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/compiler.py", line 1086, in translate_stage
    return interp.interpret(bytecode)
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/interpreter.py", line 97, in interpret
    self.cfa.run()
  File "/home/yanxon/anaconda3/lib/python3.7/site-packages/numba/controlflow.py", line 664, in run
    raise UnsupportedError(msg, loc=l)
numba.errors.UnsupportedError: Failed in nopython mode pipeline (step: analyzing bytecode)
Use of unsupported opcode (SETUP_EXCEPT) found

File "pyxtal_ff/descriptors/SO4.py", line 368:
def Exponent(Rij, Rc):
    <source elided>
    x = Rij/Rc
    try:
    ^

"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "so4-NN.py", line 31, in <module>
    ff.run(mode='train', TrainData=TrainData, TestData=TestData)
  File "/home/yanxon/Desktop/PyXtal_FF/pyxtal_ff/__init__.py", line 255, in run
    testDB.store(TestData, self._descriptors, True, self.path+'ase.db')
  File "/home/yanxon/Desktop/PyXtal_FF/pyxtal_ff/utilities/__init__.py", line 97, in store
    self.add(function, data)
  File "/home/yanxon/Desktop/PyXtal_FF/pyxtal_ff/utilities/__init__.py", line 133, in add
    for i, d in enumerate(p.imap_unordered(func, _data)):
  File "/home/yanxon/anaconda3/lib/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
numba.errors.UnsupportedError: Failed in nopython mode pipeline (step: analyzing bytecode)
Use of unsupported opcode (SETUP_EXCEPT) found

File "pyxtal_ff/descriptors/SO4.py", line 368:
def Exponent(Rij, Rc):
    <source elided>
    x = Rij/Rc
    try:
    ^
```

**Comment by David-Zagaceta** [comment]
@yanxon You may need to update numba, I am encountering no issues.

**Comment by yanxon** [comment]
@David-Zagaceta 

Yes. You're right. It seems like numba 0.44 is depreciated. I believe @qzhu2017 had fix this too in e00a053c696f755529bededdd9b7a00f7931d221.

I'm going to close this issue.
